go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -840,17 +873,8 @@ func (be *BuiltinBackupEngine) backupFile(ctx context.Context, params BackupPara
 		return vterrors.Wrap(err, "cannot copy data")
 	}
 
-	// Close gzip to flush it, after that all data is sent to writer.
```

Comment:
Is the reason for moving this just to be more consistent with where this code is handled in `restoreFile`. I am all for this change just wanted to understand the reasoning.

Reply:
The reason for moving it is to be more `golang` idiomatic: we create the `compressor` in the `if backupStorageCompress` block, and so this is where we should create the deferred `Close()` function, rather than two blocks down the road.